llvm/lib/Target/AMDGPU/GCNSchedStrategy.cpp

```diff
@@ -1716,6 +1712,11 @@ bool PreRARematStage::isTriviallyReMaterializable(const MachineInstr &MI) {
   if (!DAG.TII->isTriviallyReMaterializable(MI))
     return false;
 
+  // Even though TargetInstrInfo::isReallyTriviallyReMaterializable already
+  // ensures that the instruction has no virtual register uses,
+  // SIInstrInfo::isReallyTriviallyReMaterializable may consider an instruction
+  // rematerializable and return before calling its parent's method, so we need
+  // to double-check here.
```

Comment:
Seems like a bug? All of these rematerialize functions are confusing

Reply:
It is a bit confusing to me as well. Our `SIInstrInfo` override considers that rematerializing VOP{1,2,3} instructions with virtual register uses is trivial, which I don't really agree with. Removing the override in favor of its parent's method breaks many unit tests however, so if that's ok I will leave this as is for now and investigate this further in the future.

Reply:
I think 08d7eec06e8cf5c15a96ce11f311f1480291a441 never got finished / reapplied